test/integration/QueueConsumer-test.js

```diff
@@ -0,0 +1,145 @@
+
+  const channel = await connection.createChannel()
+
+  console.log('creating consumer with queueName')
+
+  const consumer = new QueueConsumer({
+    queueName,
+    connection,
+    prefetchCount: TEST_PREFETCH_LIMIT
+  })
+
+  await consumer.start()
+
+  t.context = {
+    queueName,
+    connection,
+    channel,
+    consumer
+  }
+})
+
+test.afterEach('clean up connection and channel', async (t) => {
+  const { consumer, channel, connection } = t.context
+
+  try {
+    await consumer.stop()
+    await channel.close()
+    await connection.close()
+  } catch (err) {
+    console.log('closed', err)
+  }
+})
+
+test('should be able to receive messages published on the queue', async (t) => {
+  const testMessage = { test: 1 }
+  const { queueName, consumer, channel } = t.context
+
+  await Promise.all([
+    waitForEvent(consumer, 'message', (message) => {
+      return message.content.toString() === JSON.stringify(testMessage)
+    }),
+    channel.sendToQueue(queueName, Buffer.from(JSON.stringify(testMessage)))
+  ])
+
+  t.pass()
+})
+
+test('should be able to acknowledge a message', async (t) => {
+  const testMessage = { test: 1 }
+  const { queueName, consumer, channel } = t.context
+
+  await channel.sendToQueue(queueName, Buffer.from(JSON.stringify(testMessage)))
+  const message = await waitForEvent(consumer, 'message')
+
+  await consumer.acknowledgeMessage(message)
+
+  t.pass()
+})
+
+test('should be able to reject a message and receive it again', async (t) => {
+  const testMessage = { test: 2 }
+  const { queueName, consumer, channel } = t.context
+
+  await channel.sendToQueue(queueName, Buffer.from(JSON.stringify(testMessage)))
+  const message = await waitForEvent(consumer, 'message', (message) => {
+    return message.content.toString() === JSON.stringify(testMessage)
+  })
+
+  await Promise.all([
+    waitForEvent(consumer, 'message', (message) => {
+      return message.content.toString() === JSON.stringify(testMessage)
+    }),
+    consumer.rejectMessage(message)
+  ])
+
+  t.pass()
+})
+
+test('should close the channel when when stopping the consumer', async (t) => {
+  const { consumer } = t.context
+
+  await Promise.all([
+    waitForEvent(consumer._channel, 'close'),
+    consumer.stop()
+  ])
+
+  t.pass()
+})
+
+test('should enforce the prefetch limit on the consumer\'s channel', async (t) => {
+  t.plan(1)
+  const { channel, consumer, queueName } = t.context
+  const timeoutError = new Error('timeout')
+
+  // send enough message to hit the upper limit
+  for (let i = 0; i < TEST_PREFETCH_LIMIT; i++) {
+    const testMessage = JSON.stringify({ message: i })
+    await Promise.all([
+      waitForEvent(consumer, 'message', (message) => {
+        return message.content.toString() === testMessage
+      }),
+      channel.sendToQueue(queueName, Buffer.from(testMessage))
+    ])
+  }
+
+  const lastMessage = JSON.stringify({ lastMessage: true })
+
+  try {
+    await Promise.race([
```

Comment:
I originally did this because I wasn't sure if pulling `bluebird` in was a good idea, since it doesn't play well with `async/await` syntax. I decided to pull it in anyway, but it should only be used for some of the convenience functions it supplies (ex. `Promise(...).timeout(..)`, `Promise.delay(...)`, etc.)